### DISCUSSION TITLE
How to use cartographicToSpatialFromGcs

### DISCUSSION THREAD

**Discussion by markstefanchuk** [iTwin/itwinjs-core#2466] [Q&A] [answered]
I am trying to convert lat/lng/alt to the imodel gcs so I can place a marker at that position. Is the following the correct way to use the function cartographicToSpatialFromGcs? When I try to log it to the console I get the following...

![image](https://user-images.githubusercontent.com/9969975/136714723-7cf15fca-0837-4112-9f9b-e8ebf1e9c575.png)

Here is the code snippet...
```ts       
 const cart = new Cartographic(dlng, dlat, dalt);
const origin=_imodel.cartographicToSpatialFromGcs(cart);
console.log(origin);
```

**Comment by bbastings** [comment, marked as answer]
The values for latitude and longitude for a Cartographic location are radians not degrees.

export class Cartographic implements CartographicProps {
  /**
   * @param longitude longitude, _in radians_.
   * @param latitude latitude, _in radians_.
   * @param height The height, in meters, above the ellipsoid.
   */
  private constructor(public longitude: number = 0, public latitude: number = 0, public height: number = 0) { }

Use the static Cartographic.fromDegrees method instead.

**Reply by markstefanchuk** [reply]
Thanks Brien,

Follow-up. The fromDegrees was important, but I also discovered that with cartographicToSpatialFromGcs, that the location passed actually be in the region. In my case, was passing a +ve value for longitude, but it should have been -ve.
